Module for i18n routing

njam opened the pull request:
This is a draft for a module to automatically generate language-specific routes when using vue-i18n. Given a page "foo" it will generate routes for each language, plus a language-agnostic redirect page:
- **/en/foo**: English version
- **/de/foo**: German version
- **/foo**: Will detect the user's language on the *client side* and redirect to the appropriate language-URL.

I can see quite a demand and a lot of discussions about this topic: https://github.com/nuxt/nuxt.js/issues/2145, https://github.com/nuxt/nuxt.js/issues/1884, https://github.com/nuxt/nuxt.js/issues/1942, https://github.com/nuxt/nuxt.js/issues/1879, https://github.com/nuxt/nuxt.js/issues/112#issuecomment-340574312, https://github.com/paulgv/nuxt-i18n-routing.
Maybe Nuxt could provide a canonical way of generating localized routes with a module?

I'm attaching this code just to start the conversation. It's not as a generic solution as it would need to be yet.
If you think this is a bad idea, or there's another better way to solve this problem, please feel free to close this PR.
Either way, I'm happy for any input and feedback!

cc @christopheschwyzer

pi0 commented:
@njam Woow thanks. Would you please create a new respository with [official module template](https://github.com/nuxt-community/module-template)? We can then move it into nuxt-community when was stable.

paulgv commented:
HI @njam, thanks for the module! I did something similar with support for a default language (no locale prefix in the URL): https://github.com/paulgv/nuxt-i18n

njam commented:
I moved this code to a separate repo at https://github.com/njam/nuxt-i18n-module and added some tests.
@paulgv  cool, thanks for the link! I will see if I can use your module for my use case and if any adjustments are needed. Would be great if there was a single module that covers the common use cases!
